### DISCUSSION TITLE
Adding support for tsconfig paths results in completely blank output page

### DISCUSSION THREAD

**Discussion by Quas94** [storybookjs/storybook#16025] [Help] [answered]
Hey all, I've got a project which uses tsconfig paths for cross-package importing. My directory structure is as follows, and my source code in `/root/client/src/*.ts` imports from `/root/lib/src/*` with `import X from '@lib/lib_file.ts'`

```
/root
  /client
    package.json
    tsconfig.json
    /.storybook
      main.js
    /src
  /lib
    package.json
    tsconfig.json
    /src
      lib_files.ts
```

My `/client/tsconfig.json` contains:

```
{
  // ...
  "compilerOptions": {
    "outDir": "./dist/",
    "baseUrl": "./src",
    "paths": {
      "@lib/*": ["../../lib/src/*"], // relative to baseUrl
    },
    "lib": [ "dom" ],
    "jsx": "react",
  },
  // ...
}
```

And finally, my `/.storybook/main.js`, to which I added the `tsconfig-paths-webpack-plugin` to resolve my aliases:

```
const path = require('path');
const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');

module.exports = {
  "stories": [
    "../src/**/*.stories.mdx",
    "../src/**/*.stories.@(js|jsx|ts|tsx)"
  ],
  "addons": [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    'storybook-css-modules-preset',
  ],
  "webpackFinal": async (config) => {
    config.resolve.modules = [
      ...(config.resolve.modules || []),
      path.resolve(__dirname, "../src"),
    ];
    config.resolve.roots = [path.resolve(__dirname, '../dist')];

    // JUST ADDED THIS
    config.resolve.plugins = [
      ...(config.resolve.plugins || []),
      new TsconfigPathsPlugin({
        extensions: config.resolve.extensions,
      }),
    ];
    //
    return config;
  },
}
```

Adding the `config.resolve.plugins` section to support the paths results in my Storybook building without any issues, but showing a completely blank page when I navigate to `localhost:6006`. There are no console errors and the HTML looks like this:

```
<html><head></head><body></body></html>
```

I also tried the alternative approach to handle the alias with the following, but the exact same thing occurs with the empty Storybook page.

```
    config.resolve.alias = {
      ...config.resolve.alias,
      '@lib': path.resolve(__dirname, '../../lib/src'),
    };
```

The `TsconfigPathsPlugin`/alias resolution do seem to be working on some level, however, since without their addition, Storybook fails to build, throwing errors about failing to locate `@lib/my_file`.

The completely empty output page has got my scratching my head, I have no idea where to start looking. Could anyone point me in the right direction? Thanks!

**Comment by winkerVSbecks** [comment, marked as answer]
Does storybook work if you remove your custom webpack config? I think that might be the source of the problem. For example, you should be using `config.plugins.push` instead of overriding `config.plugins`

**Reply by Quas94** [reply]
Thanks for the reply winkerVSbecks!

It turned out that the config I've posted above is actually fine. I was poking around some more upon reading your reply, and discovered that even when I rolled back to a commit in which Storybook was working without any issue earlier, I was running into the same blank page issue.

A simply `rm -rf node_modules && yarn install` did the trick. I made a mistake when initially trying to configure the `TsconfigPathsPlugin` (I created a `.storybook/webpack.config.js` file alongside my `.storybook/main.js` - oops!) and it appears that somehow corrupted my Storybook installation in the project, so that even after I fixed that issue, it kept giving me the blank page.

**Reply by winkerVSbecks** [reply]
tldr: add webpack config in `.storybook/main.js` and not in `.storybook/webpack.config.js`
